src/core/container.jl

```diff
@@ -219,3 +217,29 @@ function resample!(
 
     pc
 end
+
+struct ResampleWithESSThreshold{R, T<:Real}
```

Comment:
Maybe `T<:AbstractFloat`?

Reply:
I guess there's no harm in keeping it a bit more general.